cache/image_deleter.go

```diff
@@ -0,0 +1,56 @@
+// Package cache provides functionalities around the cache
+package cache
+
+import (
+	"github.com/buildpacks/imgutil"
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/log"
+)
+
+//go:generate mockgen -package testmock -destination ../testmock/image_deleter.go github.com/buildpacks/lifecycle/cache ImageDeleter
+
+// ImageDeleter defines the methods available to delete and compare cached images
+type ImageDeleter interface {
+	DeleteImage(image imgutil.Image)
+	ImagesEq(oldImage imgutil.Image, newImage imgutil.Image) (bool, error)
```

Comment:
Uhm, yeah I guess we can make it available outside of the interface signature but I need to think what would be the implication to put the `ImagesEq` function inside the `DeleteImage` or a wrapper method to do both (something I thought to do as some point) as it does something different than just deleting the image. 🤔 
But to do so, I have to change how I implemented it inside the `image_cache.Commit`method since it uses both methods.
Considering that this component is quite well separated I didn't think to change the signature but now that you are pointing this out could have sense

Reply:
I'm thinking something like
```
type ImageDeleter interface {
	DeleteImageIfExists(image imgutil.Image)
}
```
...and not exposing any of `ImagesEq` to the `cache` package

Reply:
Thank you for pointing this out @natalieparellano !
Indeed the ImagesEq was breaking the [Interface Segretation Principle](https://en.wikipedia.org/wiki/Interface_segregation_principle) and [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle) principle.
I changed the implementation following the Dependency Inversion Principle maintaining a clear separation of responsibilities allowing more flexibility in choosing different image comparing strategies without affecting the `ImageDeleter` interface using the `ImageComparer` component.

As a future improvement we can move this component in the `imageutil` library.

Let me know what do you think 😄